src/main/java/com/spotify/google/cloud/pubsub/client/Puller.java

```diff
@@ -224,6 +224,9 @@ private void pullBatch() {
               outstandingMessages.decrementAndGet();
               LOG.error("Message handler threw exception", e);
               continue;
+            } catch (Error e) {
```

Comment:
My thought was that re-throwing it below would leave whatever error handling is appropriate in place. So I don't think it'll swallow the errors. But upon further thought it won't necessarily decrement appropriately - if it gets 100 messages and the 3rd one in the loop re-throws something it'll already have incremented by 100 but will only decrement for the ones prior to and including the first error.

So this needs to treat all throwables the same way, or work out a different number to decrement by when breaking out of the for loop, or only increment the outstanding messages count as it starts each handler.

I don't think it's safe to catch and swallow all `Error`s, even if the swallowing is not silent - eg https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadDeath.html is deprecated but if anything did trigger it it would be bad to not rethrow it.

So maybe it's important to do one of the other approaches.

Reply:
This code is running in a [terminal `whenComplete` completion stage](https://github.com/spotify/async-google-pubsub-client/blob/master/src/main/java/com/spotify/google/cloud/pubsub/client/Puller.java#L198), any exceptions thrown by [the invocation of this code](https://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/b5c6ff776bee/src/share/classes/java/util/concurrent/CompletableFuture.java#l760), including any `Error` types, will be [silently swallowed](https://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/b5c6ff776bee/src/share/classes/java/util/concurrent/CompletableFuture.java#l765).